x/poolmanager/router_test.go

```diff
@@ -1530,3 +1588,419 @@ func (suite *KeeperTestSuite) setupPools(poolType types.PoolType, poolDefaultSwa
+			Pools: []types.SwapAmountInRoute{
+				{
+					PoolId:        fooBarPoolId,
+					TokenOutDenom: bar,
+				},
+				{
+					PoolId:        barUosmoPoolId,
+					TokenOutDenom: uosmo,
+				},
+				{
+					PoolId:        bazUosmoPoolId,
+					TokenOutDenom: baz,
+				},
+			},
+			TokenInAmount: sdk.NewInt(twentyFiveBaseUnitsAmount.Int64() * 3),
+		}
+
+		priceImpactThreshold = sdk.NewInt(97866545)
+	)
+
+	tests := map[string]struct {
+		isInvalidSender   bool
+		routes            []types.SwapAmountInSplitRoute
+		tokenInDenom      string
+		tokenOutMinAmount sdk.Int
+
+		// This is an estimate because we do not care about
+		// testing exact values at this level of abstraction.
+		// The math should be tested per-module.
+		// We keep this assertion to make sure that the
+		// actual result is within a reasonable range.
+		expectedTokenOutEstimate sdk.Int
+
+		expectError error
+	}{
+		"valid solo route one hop": {
+			routes:            defaultSingleRouteOneHop,
+			tokenInDenom:      foo,
+			tokenOutMinAmount: sdk.OneInt(),
+
+			expectedTokenOutEstimate: twentyFiveBaseUnitsAmount,
+		},
+		"valid solo route multi hop": {
+			routes:            []types.SwapAmountInSplitRoute{defaultSingleRouteTwoHops},
+			tokenInDenom:      foo,
+			tokenOutMinAmount: sdk.OneInt(),
+
+			expectedTokenOutEstimate: twentyFiveBaseUnitsAmount,
+		},
+		"valid split route multi hop": {
+			routes: []types.SwapAmountInSplitRoute{
+				defaultSingleRouteTwoHops,
+				defaultSingleRouteThreeHops,
+			},
+			tokenInDenom:      foo,
+			tokenOutMinAmount: sdk.OneInt(),
+
+			// 1x from single route two hops and 3x from single route three hops
+			expectedTokenOutEstimate: twentyFiveBaseUnitsAmount.MulRaw(4),
+		},
+
+		"valid split route multi hop with price impact protection that would fail individual route if given per multihop": {
+			routes: []types.SwapAmountInSplitRoute{
+				defaultSingleRouteTwoHops,
+				defaultSingleRouteThreeHops,
+			},
+			tokenInDenom: foo,
+			// equal to the expected amount
+			// every route individually would fail, but the split route should succeed
+			tokenOutMinAmount: priceImpactThreshold,
+
+			// Note: this value was taken from the actual result
+			// and not manually calculated. This is acceptable
+			// for this test because we are not testing the math
+			// but the routing logic.
+			expectedTokenOutEstimate: priceImpactThreshold,
+		},
+
+		"error: price impact protection triggerred": {
+			routes: []types.SwapAmountInSplitRoute{
+				defaultSingleRouteTwoHops,
+				defaultSingleRouteThreeHops,
+			},
+			tokenInDenom: foo,
+			// one greater than expected amount
+			// every route individually would fail, but the split route should succeed
+			tokenOutMinAmount: priceImpactThreshold.Add(sdk.OneInt()),
+
+			// Note: this values were taken from the actual result
+			// and not manually calculated. This is acceptable
+			// for this test because we are not testing the math
+			// but the routing logic.
```

Comment:
nit: I think its fine to put this comment where we define the var and remove it from under the tests themselves